### PR TITLE
[Feat]: Add function to label a long matrix with II, IE, EI, and EE

### PR DESCRIPTION
**Pull request by BenTaylor-TfN**
# Changes

Added new function `mark_internal_external`
Added tests for new function

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

----
📚 Documentation preview 📚: https://caftoolkit--215.org.readthedocs.build/en/215/

### CODE REVIEW

File: tests/pandas_utils/test_matrices.py [hunk 1]
```diff
@@ -608,3 +608,251 @@ def test_comparison_tlds(
         )
 
         assert len(comparison["TLD comparison"].columns) == 6  # noqa: PLR2004
+
+
+class TestMarkInternalExternal:
```
Comment (MattBuckley-TfN) [inline tests/pandas_utils/test_matrices.py]: Tests look like they cover everything although I think they'd probably be simpler if you used `pd.testing.assert_frame_equal` because it would probably handle the few different tests in one go

**Review by MattBuckley-TfN — CHANGES_REQUESTED** [review]
One suggestion for using pandas `NA` instead of text and then a single pylint issue to fix. Other than that looks good.

**Review by MattBuckley-TfN — APPROVED** [review]
LGTM
